lib/task-queue/Queue.js

```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+let Queue = function() {
+  this._queue = [];
+};
+
+Queue.prototype = {
+  enqueue(data) {
+    if (data === null) {
+      return;
+    }
+    this._queue.push(data);
+  },
+
+  dequeue() {
+    return this._queue.length ? this._queue.pop() : null;
+  },
+
+  peek() {
+    return this._queue.length ? this._queue[0] : null;
```

Comment:
`peek` should in fact return the element that would be popped. It allows one to have a "preview" of the data.

a proper implementation should be:

```javascript
  peek() {
    return this._queue.length ? this._queue[this._queue.length-1] : null;
  }
```